be/extmap.c

```diff
@@ -267,32 +374,69 @@ m0_be_emap_init(struct m0_be_emap *map, struct m0_be_seg *db)
 	m0_buf_init(&map->em_val_buf, &map->em_rec, sizeof map->em_rec);
 	emap_key_init(&map->em_key);
 	emap_rec_init(&map->em_rec);
-	m0_be_btree_init(&map->em_mapping, db, &be_emap_ops);
+
+	if (check) {
+		M0_ALLOC_PTR(map->em_mapping);
+		if (map->em_mapping == NULL)
+			M0_ASSERT(0);
+
+		rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op,
+						m0_btree_open(&map->em_mp_node,
+								sizeof map->em_mp_node,
+								map->em_mapping, db,
+								&b_op));
+		M0_ASSERT(rc == 0);
+	}
 	map->em_seg = db;
 	map->em_version = 0;
 	m0_format_footer_update(map);
 }
 
-M0_INTERNAL void m0_be_emap_fini(struct m0_be_emap *map)
+M0_INTERNAL void m0_be_emap_fini(struct m0_be_emap *map, bool check)
```

Comment:
Change the name of the flag as mentioned above.

Reply:
Done.